lib/rubocop/cop/mixin/target_solidus_version.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for checking target solidus version.
+    module TargetSolidusVersion
+      DEFAULT_SOLIDUS_VERSION = '3.0'
+      RUBOCOP_SOLIDUS_HOME = File.realpath(File.join(File.dirname(__FILE__), '..', '..', '..', '..')).freeze
+      DOTFILE = '.rubocop.yml'
+
+      def minimum_supported_solidus_version?(version)
+        version <= solidus_version
+      end
+
+      private
+
+      def solidus_version
+        @solidus_version ||= config.exist? ? YAML.load_file(config)['AllCops']['TargetSolidusVersion'] : DEFAULT_SOLIDUS_VERSION
```

Comment:
Fetches solidus version from the yaml file or else defaults to solidus version 3.0 (example default, can be changed as per requirement in the final edit)